server/routes/message.router.js

```diff
@@ -0,0 +1,124 @@
+const express = require('express');
+const pool = require('../modules/pool');
+const router = express.Router();
+const {
+  rejectUnauthenticated,
+} = require('../modules/authentication-middleware');
+
+router.get('/all', rejectUnauthenticated, (req, res) => {
+  // GET ROUTE - Gets all messages for currently logged-in user
+  const user = req.user.id;
+  const queryText = `
+    SELECT DISTINCT 
+      ON(GREATEST("fromUser", "toUser"), LEAST("fromUser", "toUser")) 
+      GREATEST("fromUser", "toUser"), 
+      LEAST("fromUser", "toUser"), 
+      "date" AS "maxDate", 
+      "message",
+      "users"."firstName",
+      "users"."lastName",
+      "vendors"."companyName",
+      "vendors"."profilePic"
+    FROM "messages"
+    JOIN "users"
+      ON "messages"."fromUser" = "users"."id" OR "messages"."toUser" = "users"."id"
+    RIGHT OUTER JOIN "vendors"
+      On "users"."id" = "vendors"."vendorUserId"
+    WHERE "fromUser" = $1
+      OR "toUser" = $1
+    ORDER BY GREATEST("fromUser", "toUser"), LEAST("fromUser", "toUser"), "maxDate" ASC;
+  `;
+
+  pool
+    .query(queryText, [user])
+    .then((dbRes) => {
+      console.log('SERVER - GET - at /api/message/all - received a response');
+      res.send(dbRes.rows);
+    })
+    .catch((err) => {
+      console.error(
+        'SERVER - GET at /api/message/all - an error occurred',
+        err
+      );
+      res.sendStatus(500);
+    });
+});
+
+router.get('/:id', rejectUnauthenticated, (req, res) => {
+  // GET ROUTE - Gets Messages between two users
+  const fromUser = req.user.id;
+  console.log('fromUser', fromUser);
+  const toUser = req.params.id;
```

Comment:
🤔  is this getting a message by id? Or by a user Id?

Reply:
Getting messages where the toUser id is clicked like initiating a chat, and the fromUser is the currently logged in user.